onedal/linear_model/linear_model.py

```diff
@@ -198,8 +161,43 @@ def __init__(
         super().__init__(fit_intercept=fit_intercept, copy_X=copy_X, algorithm=algorithm)
 
     def fit(self, X, y, queue=None):
-        return super()._fit(X, y, _backend.linear_model.regression, queue)
+        module = _backend.linear_model.regression
 
-    def predict(self, X, queue=None):
-        y = super()._predict(X, _backend.linear_model.regression, queue)
-        return y
+        policy = self._get_policy(queue, X, y)
+
+        X_loc, y_loc = X, y
+        if not isinstance(X, np.ndarray):
+            X_loc = np.asarray(X)
+
+        dtype = get_dtype(X_loc)
+        if dtype not in [np.float32, np.float64]:
+            dtype = np.float64
+            X_loc = X_loc.astype(dtype, copy=self.copy_X)
+
+        y_loc = np.asarray(y_loc).astype(dtype=dtype)
+
+        # Finiteness is checked in the sklearnex wrapper
+        X_loc, y_loc = _check_X_y(X_loc, y_loc, force_all_finite=False, accept_2d_y=True)
+
+        self.n_features_in_ = _num_features(X_loc, fallback_1d=True)
+
+        X_loc, y_loc = _convert_to_supported(policy, X_loc, y_loc)
+        params = self._get_onedal_params(get_dtype(X_loc))
+        X_table, y_table = to_table(X_loc, y_loc)
+
+        hparams = get_hyperparameters("linear_regression", "train")
+        if hparams is not None and not hparams.is_default:
+            result = module.train(policy, params, hparams.backend, X_table, y_table)
+        else:
```

Comment:
How this linear regression refactoring/updating related with incremental impl?
Each changes on previous estimator shoudl be done on separate branch with separate validation , perf validation

Reply:
The methods in BaseLinearRegression class should be intersection of methods used in LinearRegression and IncrementalLinearRegression. Otherwise, this class is redundant. API for the estimator remains the same, the code also remains the same, methods just moved from one class to another. I don't think it is worth to open new PR here

Reply:
Thank you for clarification!
In any way this will affect on other LinearRegression estimator that is already on main branch.
I worth to make this common changes first in seperate branch, validate it via tests, perf and then merge.
After this increment better to add IncrementalLR

Reply:
My point is that there is no affect on executed code, so, perf is not expected to be changed, tests for the estimator are still green, and considering our capacity for review PRs it might be worth to leave it as is. Moreover, the changes are necessary to introduce IncrementalLinearRegression estimator, that's why it makes sense to keep it in this PR as well.

Reply:
I don't understand why are you that sure. The only thing that we have to base is numbers and tests validations.
Doing this changes incrementally is better for review and integration and safety

Reply:
Test validations are here. About perf validations, after the changes the interpreter goes through exactly the same rows and instructions, so, using common sense one can conclude that perf must remain exactly the same.

Reply:
I see a lot of common code for fit and partial_fit, so worth to have common fit for this.
If you think it will more time to impl, then you can consider to leave.